src/ngx_http_lua_regex.c

```diff
@@ -1923,11 +1927,51 @@ ngx_http_lua_ngx_re_sub_helper(lua_State *L, unsigned global)
 
 
 void
+ngx_http_lua_set_jit_stack_size(int size)
+{
+    ngx_http_lua_main_conf_t    *lmcf;
+
+    if (size < NGX_LUA_RE_MIN_JIT_STACK_SIZE) {
+        size = NGX_LUA_RE_MIN_JIT_STACK_SIZE;
+    }
+
+    lmcf = ngx_http_cycle_get_module_main_conf(ngx_cycle,
+                                               ngx_http_lua_module);
+    // pcre_jit_stack_free(lmcf->jit_stack); // Y U NO WORK?
+    lmcf->jit_stack = pcre_jit_stack_alloc(NGX_LUA_RE_MIN_JIT_STACK_SIZE,
+                                           size);
```

Comment:
We should handle the case that `pcre_jit_stack_alloc` returns a `NULL` when running out of memory.